service/history/workflow/transaction_impl.go

```diff
@@ -727,6 +730,10 @@ func emitCompletionMetrics(
 }
 
 func operationPossiblySucceeded(err error) bool {
+	if err == consts.ErrConflict {
+		return false
+	}
```

Comment:
i believe the logic should get rid of `consts.ErrConflict` in the long term

Reply:
There is a TODO for this (not by me), but I am not ready for this. Yet.